docs/dev/table/sqlClient.md

```diff
@@ -0,0 +1,538 @@
+
+
+{% top %}
+
+Table Sources
+-------------
+
+Sources are defined using a set of [YAML properties](http://http://yaml.org/). Similar to a SQL `CREATE TABLE` statement you define the name of the table, the final schema of the table, connector, and a data format if necessary. Additionally, you have to specify it's type (source, sink, or both).
+
+{% highlight yaml %}
+name: MyTable     # required; string representing the table name
+type: source      # required; currently only 'source' is supported
+schema: ...       # required; final table schema
+connector: ...    # required; connector configuration
+format: ...       # optional; format that depends on the connector 
+{% endhighlight %}
+
+**Note:** Not every combination of connector and format is supported. Internally, your YAML file is translated into a set of string-based properties by which the SQL Client tries to resolve a matching table source. If a table source can be resolved also depends on the JAR files available in the classpath.
+
+The following example shows an environment file that defines a table source reading JSON data from Apache Kafka. All properties are explained in the following subsections. 
+
+{% highlight yaml %}
+tables:
+  - name: TaxiRides
+    type: source
+    schema:
+      - name: rideId
+        type: LONG
+      - name: lon
+        type: FLOAT
+      - name: lat
+        type: FLOAT
+      - name: rowTime
+        type: TIMESTAMP
+        rowtime:
+          timestamps:
+            type: "from-field"
+            from: "rideTime"
+          watermarks:
+            type: "periodic-bounded"
+            delay: "60000"
+      - name: procTime
+        type: TIMESTAMP
+        proctime: true
+    connector:
+      property-version: 1
+      type: kafka
+      version: 0.11
+      topic: TaxiRides
+      startup-mode: earliest-offset
+      properties:
+        - key: zookeeper.connect
+          value: localhost:2181
+        - key: bootstrap.servers
+          value: localhost:9092
+        - key: group.id
+          value: testGroup
+    format:
+      property-version: 1
+      type: json
+      schema: "ROW(rideId LONG, lon FLOAT, lat FLOAT, rideTime TIMESTAMP)"
+{% endhighlight %}
+
+The resulting schema of the `TaxiRide` table contains most of the fields of the JSON schema. Furthermore, it adds a rowtime attribute `rowTime` and processing-time attribute `procTime`. Both `connector` and `format` allow to define a property version (which is currently version `1`) for future backwards compatibility.
+
+{% top %}
+
+### Schema Properties
+
+The schema allows for describing the final appearence of a table. It specifies the final name, final type, and the origin of a field. For instance, the origin of a field might be important if time attributes need to be declared or the name of the field should differ from the input format.
```

Comment:
1. typo: "appearence" -> "appearance"
2. I do not understand the example: "For instance, the origin of a field might be important if time attributes need to be declared or the name of the field should differ from the input format."